nncf/tensorflow/quantization/initializers/init_range.py

```diff
@@ -0,0 +1,157 @@
+from itertools import islice
+import math
+
+import tensorflow as tf
+
+from nncf.common.quantization.initialization.range import RangeInitParams
+from nncf.common.quantization.initialization.range import RangeInitConfig
+from nncf.common.quantization.structs import QuantizerGroup
+from nncf.common.utils.progress_bar import ProgressBar
+from nncf.common.utils.helpers import should_consider_scope
+from nncf.tensorflow.layers.custom_objects import NNCF_QUANTIZATION_OPERATONS
+from nncf.tensorflow.layers.wrapper import NNCFWrapper
+from nncf.tensorflow.layers.data_layout import get_channel_axis
+from nncf.tensorflow.layers.operation import InputType
+from nncf.tensorflow.quantization.layers import FakeQuantize
+from nncf.tensorflow.quantization.initializers.collectors import MinMaxStatisticCollector
+from nncf.tensorflow.quantization.initializers.collectors import MeanMinMaxStatisticsCollector
+from nncf.tensorflow.quantization.initializers.collectors import MedianMADStatisticCollector
+from nncf.tensorflow.quantization.initializers.collectors import PercentileStatisticCollector
+from nncf.tensorflow.quantization.initializers.collectors import MeanPercentileStatisticCollector
+
+
+class TFRangeInitParams(RangeInitParams):
+    def get_max_num_init_steps(self) -> int:
+        steps = []
+        if self.global_init_config is not None:
+            steps.append(self.global_init_config.num_init_samples)
+        for pl_config in self.per_layer_range_init_configs:
+            steps.append(pl_config.num_init_samples)
+        batch_size = self.init_range_data_loader.batch_size
+        return math.ceil(max(steps) / batch_size)
+
+    def get_init_config_for_quantization_point(self, layer: tf.keras.layers.Layer,
+                                               input_type: str) -> RangeInitConfig:
+        if input_type == InputType.WEIGHTS:
+            node_name = layer.name
+            group = QuantizerGroup.WEIGHTS
+        else:
+            node_name = layer.name.replace('/fake_quantize', '')
+            group = QuantizerGroup.ACTIVATIONS
+        return self.get_init_config_for_scope_and_group(node_name, group)
+
+    def get_init_config_for_scope_and_group(self, node_name: str, group: QuantizerGroup) -> RangeInitConfig:
+        matches = []  # type: List[RangeInitConfig]
+        for pl_config in self.per_layer_range_init_configs:
+            if should_consider_scope(node_name,
+                                     ignored_scopes=pl_config.ignored_scopes,
+                                     target_scopes=pl_config.target_scopes):
+                if group == pl_config.target_group or pl_config.target_group is None:
+                    matches.append(RangeInitConfig(pl_config.init_type, pl_config.num_init_samples,
+                                                   pl_config.init_type_specific_params))
+        if len(matches) > 1:
+            raise ValueError("Location {} matches more than one per-layer initialization parameter "
```

Comment:
Could you please use `'` instead of `"`?

Reply:
Thanks, fixed!